Python/optimizer_symbols.c

```diff
@@ -332,13 +431,60 @@ _Py_uop_sym_truthiness(_Py_UopsSymbol *sym)
     return -1;
 }
 
+static JitOptSymbol *
+allocation_base(JitOptContext *ctx)
+{
+    return ctx->t_arena.arena;
+}
+
+JitOptSymbol *
+_Py_uop_sym_new_tuple(JitOptContext *ctx, int size, JitOptSymbol **args)
+{
+    JitOptSymbol *res = sym_new(ctx);
+    if (res == NULL) {
+        return out_of_space(ctx);
+    }
+    if (size > 6) {
+        res->tag = JIT_SYM_KNOWN_CLASS_TAG;
+        res->cls.type = &PyTuple_Type;
+    }
```

Comment:
There are probably cases where this can be represented as a constant value (when all items are known).

Reply:
Hm, although it could get tricky because nothing owns that constant, so we have to be careful not to promote its uses to constant loads.

Reply:
Well, it can have more than 7 items... but sure, that probably isn't very common.